opentelemetry-api/tests/metrics/test_meter.py

```diff
@@ -0,0 +1,179 @@
+        self.assertTrue(Meter.create_up_down_counter.__isabstractmethod__)
+
+    def test_create_observable_counter(self):
+        """
+        Test that the meter provides a function to create a new ObservableCounter
+        """
+
+        self.assertTrue(hasattr(Meter, "create_observable_counter"))
+        self.assertTrue(Meter.create_observable_counter.__isabstractmethod__)
+
+    def test_create_histogram(self):
+        """
+        Test that the meter provides a function to create a new Histogram
+        """
+
+        self.assertTrue(hasattr(Meter, "create_histogram"))
+        self.assertTrue(Meter.create_histogram.__isabstractmethod__)
+
+    def test_create_observable_gauge(self):
+        """
+        Test that the meter provides a function to create a new ObservableGauge
+        """
+
+        self.assertTrue(hasattr(Meter, "create_observable_gauge"))
+        self.assertTrue(Meter.create_observable_gauge.__isabstractmethod__)
+
+    def test_create_observable_up_down_counter(self):
+        """
+        Test that the meter provides a function to create a new
+        ObservableUpDownCounter
+        """
+
+        self.assertTrue(hasattr(Meter, "create_observable_up_down_counter"))
+        self.assertTrue(
+            Meter.create_observable_up_down_counter.__isabstractmethod__
+        )
+
+    def test_no_repeated_instrument_names(self):
+        """
+        Test that the meter returns an error when multiple instruments are
+        registered under the same Meter using the same name.
+        """
+
+        meter = ChildMeter("name")
+
+        meter.create_counter("name")
+
+        with self.assertLogs(level=WARNING):
+            meter.create_counter("name")
+
+        with self.assertLogs(level=WARNING):
+            meter.create_up_down_counter("name")
+
+        with self.assertLogs(level=WARNING):
+            meter.create_observable_counter("name", Mock())
+
+        with self.assertLogs(level=WARNING):
+            meter.create_histogram("name")
+
+        with self.assertLogs(level=WARNING):
+            meter.create_observable_gauge("name", Mock())
+
+        with self.assertLogs(level=WARNING):
+            meter.create_observable_up_down_counter("name", Mock())
+
+    def test_same_name_instrument_different_meter(self):
+        """
+        Test that is possible to register two instruments with the same name
+        under different meters.
+        """
+
+        meter_0 = ChildMeter("meter_0")
+        meter_1 = ChildMeter("meter_1")
+
+        meter_0.create_counter("counter")
+        meter_0.create_up_down_counter("up_down_counter")
+        meter_0.create_observable_counter("observable_counter", Mock())
+        meter_0.create_histogram("histogram")
+        meter_0.create_observable_gauge("observable_gauge", Mock())
+        meter_0.create_observable_up_down_counter(
+            "observable_up_down_counter", Mock()
+        )
+
+        with self.assertRaises(AssertionError):
```

Comment:
Could this be added as an `assertNoLogs` utility method? it would be easier to read.

Reply:
Yes, but I prefer to avoid adding any new public interface, specially if that interface is not even related to telemetry.

Reply:
I was thinking this could live just in this test file (or in test/util). the name would help clarify what's being tested is all.

Reply:
Hm, tried it and implementation is not simple because we are dealing with context managers here. Not much to gain in my opinion.